### PR TITLE
fix crash due to passphrase caching after biom. authentication expired

### PR DESCRIPTION
**Pull request by agrahn**
I discovered another spot where app crash may occur due to expiration of biometric authentication while attempting to write to `EncryptedSharedPreferences`. It happens if the dialog for entering the passphrase is OK-closed after prior authentication has expired (keeping device awake by tapping on the dialog for a minute or so, then tapping OK):

https://github.com/android-password-store/Android-Password-Store/blob/7010ee85b4bdfa0c8120f15597a9471151576b2c/app/src/main/java/app/passwordstore/ui/crypto/DecryptActivity.kt#L218

The PR wraps the line in question in a `runCatching` block, such that if passphrase caching fails, the `DecryptActivity` screen is still completed and displayed. Similary, with `AutofillDecryptActivity`, decrypted credentials are now still filled in on a login form, even if caching fails.

### CODE REVIEW

File: app/src/main/java/app/passwordstore/ui/autofill/AutofillDecryptActivity.kt [hunk 1]
```diff
@@ -218,9 +216,15 @@ class AutofillDecryptActivity : BasePGPActivity() {
...
-                }
+                runCatching {
+                    if (features.isEnabled(EnablePGPPassphraseCache)) {
+                      passphraseCache.cachePassphrase(this, identifiers.first(), password)
+                      settings.edit {
+                        putBoolean(PreferenceKeys.CLEAR_PASSPHRASE_CACHE, clearCache)
+                      }
+                    }
+                  }
+                  .onFailure { e -> e.printStackTrace() }
```
Comment (msfjarvis) [inline app/src/main/java/app/passwordstore/ui/autofill/AutofillDecryptActivity.kt]:
```suggestion
                  .onFailure { e -> logcat { e.asLog() } }
```

You'll have to import `logcat.asLog`.

File: app/src/main/java/app/passwordstore/ui/crypto/DecryptActivity.kt [hunk 2]
```diff
@@ -214,13 +216,17 @@ class DecryptActivity : BasePGPActivity() {
...
+                if (authResult is BiometricResult.Success) {
+                  passphraseCache.cachePassphrase(
+                    this@DecryptActivity,
+                    gpgIdentifiers.first(),
+                    passphrase,
+                  )
+                  settings.edit { putBoolean(PreferenceKeys.CLEAR_PASSPHRASE_CACHE, clearCache) }
+                }
+              }
+              .onFailure { e -> e.printStackTrace() }
```
Comment (msfjarvis) [inline app/src/main/java/app/passwordstore/ui/crypto/DecryptActivity.kt]:
```suggestion
              .onFailure { e -> logcat { e.asLog() } }
```

Same thing here.

File: app/src/main/java/app/passwordstore/ui/crypto/DecryptActivity.kt [hunk 3]
```diff
@@ -237,8 +243,6 @@ class DecryptActivity : BasePGPActivity() {
     val outputStream = ByteArrayOutputStream()
     val result = repository.decrypt(passphrase, identifiers, message, outputStream)
     if (result.isOk) {
-      if (features.isEnabled(EnablePGPPassphraseCache))
-        settings.edit { putBoolean(PreferenceKeys.CLEAR_PASSPHRASE_CACHE, clearCache) }
```
Comment (msfjarvis) [inline app/src/main/java/app/passwordstore/ui/crypto/DecryptActivity.kt:241]: Why are we changing this?

Comment (agrahn) [reply app/src/main/java/app/passwordstore/ui/crypto/DecryptActivity.kt:241]: I thought I move `settings.edit { ... }` to the place where the passphrase is written to the cache (now on line 226). Both operations belong together somehow. A change of this setting is only persisted if the passphrase cache could be successfully written too (they are in the same `runCatching` block). Its now at the same place in both, `DecryptActivity` and `AutofillDycryptActivity`.

File: app/src/main/java/app/passwordstore/ui/autofill/AutofillDecryptActivity.kt (hunk 1 above)
Comment (agrahn) [reply app/src/main/java/app/passwordstore/ui/autofill/AutofillDecryptActivity.kt]: Thanks, I edited the line as suggested and did the same in the other file.

**Review by msfjarvis — APPROVED** [review]

**Pull request merged**